enterprise/server/backends/distributed/distributed.go

```diff
@@ -973,7 +979,18 @@ func (c *Cache) SetMulti(ctx context.Context, kvs map[*repb.Digest][]byte) error
 }
 
 func (c *Cache) Delete(ctx context.Context, d *repb.Digest) error {
-	return status.UnimplementedError("Not yet implemented.")
+	ps := c.readPeers(d)
+	for peer := ps.GetNextPeer(); peer != ""; peer = ps.GetNextPeer() {
+		err := c.remoteDelete(ctx, peer, c.isolation, d)
+		if err != nil {
+			if status.IsNotFoundError(err) {
+				c.log.Warningf("File not found, could not be deleted: %s", err.Error())
```

Comment:
Do we care if it doesn't exist? I'm not sure it's worth a warning.

Reply:
I thought a warning might be nice if you had a typo in the file name or something, but the warning logic isn't consistent so I removed it (i.e. if the data wasn't written to a peer as expected, client shouldn't be warned about that. Also some of the cache types, like S3, don't expose whether the file existed or not before deletion)